extensions/dacpac/package.json

```diff
@@ -35,12 +35,12 @@
       "objectExplorer/item/context": [
         {
           "command": "dacFx.start",
-          "when": "connectionProvider == MSSQL && nodeType && nodeType == Database && mssql:engineedition != 11",
+          "when": "connectionProvider == MSSQL && nodeType && nodeType == Database && nodeLabel != 'master' && nodeLabel != 'model' && nodeLabel != 'msdb' && nodeLabel != 'tempdb' && mssql:engineedition != 11",
           "group": "export"
         },
         {
           "command": "dacFx.start",
-          "when": "connectionProvider == MSSQL && nodeType && nodeType == Server && mssql:engineedition != 11",
+          "when": "connectionProvider == MSSQL && nodeType && nodeType == Server && !isSystemDb && !isDefaultDb && mssql:engineedition != 11",
```

Comment:
there is a bug here, basically you want to show the menu item if user is connecting to a user database. but a user can specify their default database (by default it is master) so that when a connection is made, the connection will be to the default database. checking whether databaseName property of the connection profile is empty is not enough. 

a reliable way would be check the actual database the connection is connection after the connection is made.